File: reference/springboot/domain-usecases/src/main/java/org/example/domain/usecases/types/Cow.java
package org.example.domain.usecases.types;

import lombok.Builder;
import lombok.Getter;

import java.util.UUID;

@Getter
public class Cow extends Animal {
    private String c;

    @Builder
    public Cow(UUID id, String sound, String c) {
        super(id, sound);
        this.c = c;
    }

    @Override
    public <T> T visit(AnimalVisitor<T> visitor) {
        return visitor.visitCow(this);
    }
}

File: reference/dotnet/Domain/Company.Product.Domain.UseCases/Types/Cow.cs
using System;

namespace Company.Product.Domain.UseCases.Types
{
    public class Cow : Animal
    {
        public string C { get; }

        public Cow(Guid animalId, string sound, string c) 
            : base(animalId, sound)
        {
            C = c;
        }

        public override T Visit<T>(AnimalVisitor<T> visitor)
        {
            if (visitor is null)
            {
                throw new ArgumentNullException(nameof(visitor));
            }

            return visitor.Visit(this);
        }
    }
}